homework/index.js

```diff
@@ -29,16 +14,110 @@
+        throw new Error('Network request failed');
+      })
+      .then(res => res.json());
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function showError(err) {
+    const root = document.getElementById('root');
+
+    createAndAppend('div', root, {
+      text: 'HYF Repositories',
+      class: 'HYF-Repositories',
+      id: 'HYF-Repositories',
+    });
+    createAndAppend('div', root, {
+      text: new Error(err.message),
+      class: 'alert-error',
+    });
+  }
+
+  const renderContributors = contributor => {
+    const contributions = document.getElementById('contributions');
+    contributions.innerHTML = 'Contributors';
+    if (contributor && contributor.length > 0) {
+      for (let contribution = 0; contribution < contributor.length; contribution++) {
+        const logi = contributor[contribution].login;
+        const img = contributor[contribution].avatar_url;
+        const link = contributor[contribution].html_url;
+        const contribute = contributor[contribution].contributions;
+        const contributorsList = `<li class='right-li'><a target=_blank href=${link}><img src=${img}><br>${logi}<br>Contributions: ${contribute}</a></li>`;
+
+        contributions.innerHTML += contributorsList;
       }
+    } else {
+      contributions.innerHTML += '<li>No information available</li>';
+    }
+  };
+  function makeApp(jsonFile) {
+    jsonFile.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const hyfRepositories = createAndAppend('div', root, {
+      class: 'HYF-Repositories',
+      id: 'HYF-Repositories',
+    });
+    createAndAppend('h1', hyfRepositories, { text: 'HYF Repositories' });
+    const repositories = createAndAppend('select', hyfRepositories, { id: 'repositories' });
+    const informationBox = createAndAppend('div', root, {
+      class: 'information-box',
+      id: 'information-box',
     });
+    const repositoryInformation = createAndAppend('ul', informationBox, {
+      id: 'repository-information',
+    });
+    const contributions = createAndAppend('ul', informationBox, {
+      text: 'Contributors',
+      id: 'contributions',
+      class: 'right-ul',
+    });
+    const RepositoryName = createAndAppend('li', repositoryInformation, { text: '' });
+    const description = createAndAppend('li', repositoryInformation, { text: '' });
+    const forks = createAndAppend('li', repositoryInformation, { text: '' });
+    const updat = createAndAppend('li', repositoryInformation, { text: '' });
+
+    for (let i = 0; i < jsonFile.length; i++) {
+      createAndAppend('option', repositories, { text: jsonFile[i].name, value: i });
+    }
+
+    async function displayRepoInformation() {
```

Comment:
Another thing here - even though the logic works fine here, I think there is still some room for improvement. For example, the `makeApp` function is still quite big and is doing more than one thing. It is responsible for creating all elements for repositories, displays information in those created elements and then fetches contributors information.

I think we can split this function a bit more. For example, the parent function can be responsible for creating all elements and the dropdown for repositories. The `displayRepoInformation` function can be moved out of the `makeApp` function. As far as the `repositories.value`, `repositoryName`, `description` and other parameters are concerned, they can be passed on to this function as parameters.

Drop me a message on Slack if you have any questions around it.  🙂